clvm/SExp.py

```diff
@@ -217,7 +217,7 @@ def null(class_) -> SExp:
 
     def as_iter(self: _T_SExp) -> typing.Iterator[_T_SExp]:
         v = self
-        while not v.nullp():
+        while v.listp():
```

Comment:
```suggestion
        while v.pair is not None:
```

Reply:
At some point, I created `CLVMObject` which is supposed to be the "real" way to deal with the `Union` that is a clvm node: exactly one of `.atom` and `.pair` is `None`. I think it's better than using `.listp` (which should probably just go away).

Reply:
that was my first version of this :)